framework/merchant_api/routes.py

```diff
@@ -0,0 +1,75 @@
+from base64 import b64encode
+from typing import Dict, List
+from uuid import UUID
+
+import requests
+from flask import Blueprint, jsonify
+
+from framework.merchant_api.domain.entities.dora_product import DoraProduct
+from framework.merchant_api.domain.entities.scraped_product_offer import \
+    ScrapedProductOffer
+from framework.merchant_api.domain.enumerations.supported_merchant import \
+    SupportedMerchant
+from framework.merchant_api.infrastructure.session import create_session
+from framework.merchant_api.scrapers import coles_scraper, woolworths_scraper
+
+# TODO: Logging --- would be useful to know what the search term was that caused the error, and which merchant, etc
+
+PRODUCT_ROUTER = Blueprint("PRODUCT_ROUTER", __name__, url_prefix="/api/products")
+
+
+@PRODUCT_ROUTER.route("/search/<search_term>")
+@PRODUCT_ROUTER.route("/search/<search_term>/<start_page>")
+@PRODUCT_ROUTER.route("/search/<search_term>/<start_page>/<result_limit>")
+async def search_for_product_async(search_term: str, start_page: int = 1, result_limit: int = 10) -> List[ScrapedProductOffer]:
+    with create_session() as _Session:
```

Comment:
what is the benefit to using this session? from what i understand it maintains state between multiple requests but can't tell how that helps here

Reply:
i don't have a deep enough understanding, it's something i took from the original code i copied

what i do know is it speeds up the requests made dramatically likely due to caching (i found it was very slow without it, sometimes timed out), and it also allows you to customise the behaviour of the requests

Reply:
Interesting, i'll try to find more about it